cypress/integration/EventWorkingLists/EventWorkingListsDev/index.js

```diff
@@ -0,0 +1,307 @@
+        .should('equal', 200);
+
+    cy.get('@result')
+        .its('url')
+        .should('include', 'assignedUserMode=ANY');
+
+    cy.get('@result')
+        .its('url')
+        .should('include', 'status=ACTIVE');
+
+    cy.get('@result')
+        .its('url')
+        .should('include', 'page=1');
+
+    cy.get('@result').its('response.body.events').as('events');
+});
+
+When('you set the age filter to 10-20', () => {
+    cy.get('[data-test="event-working-lists"]')
+        .contains('Age (years)')
+        .click();
+
+    cy.get('[data-test="list-view-filter-contents"]')
+        .find('input[placeholder="Min"]')
+        .type('10');
+
+    cy.get('[data-test="list-view-filter-contents"]')
+        .find('input[placeholder="Max"]')
+        .type('20');
+});
+
+When('you apply the current filter', () => {
+    cy.route('GET', '**/events**').as('getEvents');
+
+    cy.get('[data-test="list-view-filter-apply-button"]')
+        .click();
+});
+
+Then('the age filter button should show that the filter is in effect', () => {
+    cy.get('[data-test="event-working-lists"]')
+        .contains('Age (years): 10 to 20')
+        .should('exist');
+});
+
+Then('events where age is between 10 and 20 should be retrieved from the api', () => {
+    cy.wait('@getEvents', { timeout: 40000 }).as('result');
+
+    cy.get('@result')
+        .its('status')
+        .should('equal', 200);
+
+    cy.get('@result')
+        .its('url')
+        .should('match', /filter=.*10/);
+
+    cy.get('@result')
+        .its('url')
+        .should('match', /filter=.*20/);
+
+    cy.get('@result')
+        .its('url')
+        .should('include', 'page=1');
+
+    cy.get('@result').its('response.body.events').as('events');
+});
+
+When('you click the next page buttton', () => {
+    cy.get('[data-test="dhis2-capture-search-pagination-next-page"]')
+        .should('exist');
+
+    cy.route('GET', '**/events**').as('getEvents');
+
+    cy.get('[data-test="dhis2-capture-search-pagination-next-page"]')
+        .click();
+});
+
+Then('new events should be retrieved from the api', () => {
+    cy.wait('@getEvents', { timeout: 40000 }).as('result');
+
+    cy.get('@result')
+        .its('status')
+        .should('equal', 200);
+
+    cy.get('@result').its('response.body.events').as('events');
+});
+
+Then('the list should show the second page', () => {
+    cy.get('[data-test="event-working-lists"]')
+        .contains('Page 2')
+        .should('exist');
+});
+
+When('you click the previous page button', () => {
+    cy.route('GET', '**/events**').as('getEvents');
+
+    cy.get('[data-test="dhis2-capture-search-pagination-previous-page"]')
+        .click();
+});
+
+When('you click the first page button', () => {
+    cy.route('GET', '**/events**').as('getEvents');
+
+    cy.get('[data-test="dhis2-capture-search-pagination-first-page"]')
+        .click();
+});
+
+When('you change rows per page to 50', () => {
+    cy.get('div[id="rows-per-page-selector"]')
```

Comment:
I see you chose to use an id  here. Isnt there a way to go with the `data-test` attribute?

Reply:
I was working with what I already had here. But I agree data-test would be better here. Adding a data-test attribute to this one 👍